### PR TITLE
Support zone

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

File: src/common/interface/meta.thrift [hunk 1]
```diff
@@ -826,6 +828,85 @@ struct ListIndexStatusResp {
...
+    1: binary                 group_name,
+}
+
+struct ListZonesResp {
+    1: ErrorCode        code,
+    2: common.HostAddr  leader,
+    3: list<Zone>       zones,
+}
+
+struct DrainZoneReq {
```
Comment (critical27) [inline src/common/interface/meta.thrift:873]: According to design doc, DrainZoneReq should only used in default mode?

File: src/common/interface/meta.thrift [hunk 2]
```diff
@@ -826,6 +828,85 @@ struct ListIndexStatusResp {
...
+    1: binary        group_name,
+    2: binary        zone_name,
+}
+
+struct AddGroupReq {
```
Comment (critical27) [inline src/common/interface/meta.thrift:878]: According to design doc, should we specify which group is mirror?

File: src/common/interface/meta.thrift [hunk 3]
```diff
@@ -336,23 +337,24 @@ struct AdminJobResp {
...
 struct ListSpacesReq {
 }
 
 struct ListSpacesResp {
-    1: ErrorCode code,
+    1: ErrorCode        code,
     // Valid if ret equals E_LEADER_CHANGED.
     2: common.HostAddr  leader,
-    3: list<IdName> spaces,
+    3: list<IdName>     spaces,
```
Comment (liuyu85cn) [inline src/common/interface/meta.thrift:357]: I'm not sure if we need to show zone info also.

File: src/common/interface/meta.thrift (hunk 1 above)
Comment (darionyaphet) [reply src/common/interface/meta.thrift:873]: Currently We can support replication number is greater or equal with zone number. So it's ok

File: src/common/interface/meta.thrift (hunk 2 above)
Comment (darionyaphet) [reply src/common/interface/meta.thrift:878]: I will support it in async copy module

File: src/common/interface/meta.thrift (hunk 3 above)
Comment (darionyaphet) [reply src/common/interface/meta.thrift:357]: Sure I will support it later

**Review by critical27 — APPROVED** [review]
LGTM

**Review by liuyu85cn — APPROVED** [review]
LGTM

**Pull request merged**